tls/s2n_psk.c

```diff
@@ -291,8 +317,24 @@ int s2n_offered_psk_list_choose_psk(struct s2n_offered_psk_list *psk_list, struc
         psk_params->chosen_psk = NULL;
         return S2N_SUCCESS;
     }
-
-    POSIX_GUARD_RESULT(s2n_match_psk_identity(&psk_params->psk_list, &psk->identity, &psk_params->chosen_psk));
+    
+    if (psk_params->type == S2N_PSK_TYPE_EXTERNAL) {
+        POSIX_GUARD_RESULT(s2n_match_psk_identity(&psk_params->psk_list, &psk->identity, &psk_params->chosen_psk));
+    } else {
+        struct s2n_stuffer ticket_stuffer = { 0 };
+        POSIX_GUARD(s2n_stuffer_init(&ticket_stuffer, &psk->identity));
+        POSIX_GUARD(s2n_stuffer_skip_write(&ticket_stuffer, psk->identity.size));
+        psk_list->conn->client_ticket_to_decrypt = ticket_stuffer;
+
+        POSIX_GUARD(s2n_decrypt_session_ticket(psk_list->conn));
+
+        /* If the decrypt_session_ticket method succeeds, a new PSK has been added. */
+        struct s2n_psk *decrypted_psk = NULL;
+        POSIX_GUARD_RESULT(s2n_match_psk_identity(&psk_params->psk_list, &psk->identity, &decrypted_psk));
```

Comment:
`s2n_decrypt_session_ticket` removes all previously set psks and appends the new PSK after decryption to the connection psk_params list. We should document this behavior here. Could you add a comment for it? Thanks.

Reply:
With the code-as is, it doesn't care that all other PSKs were removed. We're calling match.

Reply:
But `psk_params ` is pointing to the `conn->psk_params` and within s2n_decrypt_session_ticket we wipe the connection params_list.

Reply:
Does that effect this code in any way?

Reply:
I think previously the fact that the PSKs were removed mattered, and the comment was useful, because I was explicitly grabbing the first PSK in the list. However since we're now doing match_psk_identity it no longer matters whether or not there are more psks on the list.

Reply:
The match_psk_identity is still going to match with only the one psk at index that exists in params_list after calling s2n_decrypt_session_ticket, we are just hiding it in this `s2n_match_psk_identity` call here.

Reply:
For resumption case, the match_psk_identity is still going to match with only the one psk in the list that was added to the connection params_list after decryption (in the s2n_decrypt_session_ticket call), we are just hiding it in this s2n_match_psk_identity call here. IMO we must add this to the comments as it is not easily readable that s2n_decrypt_session_ticket removes the previously set psk_params_list.

Reply:
Would we need to change this code in any way if s2n_decrypt_session_ticket stopped deleting all the other PSKs?

Reply:
The behavior of the s2n_match_psk_identity will not change, but the fact is for resumption psk it will be match with a list of only one PSK.